Sources/Logging/Logging.swift

```diff
@@ -166,6 +166,17 @@ extension Logger {
         }
     }
 
+    /// Convenience function for adding multiple metadata items to a logger.
+    ///
+    /// - note: Logging metadata behaves as a value that means a change to the logging metadata will only affect the
+    ///         very `Logger` it was changed on.
+    @inlinable
+    public mutating func withMetadata(metadata: Logger.Metadata) {
```

Comment:
Can I propose we spell this slightly differently? 

`with` is typically used in names that provide scoped access to something (`withUnsafePointer`, `withCheckedContinuation`, `withTaskGroup` etc.) so doesn't feel quite right here (although I appreciate it is the spelling used for loggers in other languages). 

How about `addMetadata`? Could we also drop the `metadata:` label to avoid the repetition?

Reply:
Wondering if `merging` is the right term here since we merge the new values over the old values if keys are in both metadata "bags". Could we also add a doc comment for the parameter that we pass here please

Reply:
I think `merging` should be mutable and return a new `Logger`.
`merge` would be the `mutable` counterpart.

Reply:
> I think `merging` should **not** be mutable and return a new `Logger`. `merge` would be the `mutable` counterpart.

Agreed. `merge` is the correct spelling for this context. If we do this we should allow the caller to specify how to handle non-unique keys too. `Dictionary` has a spelling for this: https://developer.apple.com/documentation/swift/dictionary/merge(_:uniquingkeyswith:)-m2ub

Reply:
since `Metadata` is actually a `Dictionary`, this is already available as `logger.metadata.merge(_:uniquingKeysWith:)` 🤷‍♂️

Reply:
I do think `mergeMetadata`, paired with the dropping a required `metadata` argument label, is the clearest option here--will proceed with that.